permuta/perm.py

```diff
@@ -2271,63 +2002,48 @@ def __call__(self, value):
             >>> Perm((3, 1, 2, 0))(3)
             0
         """
-        if not isinstance(value, numbers.Integral):
-            raise TypeError("'{}' object is not an integer".format(repr(value)))
-        if not 0 <= value < len(self):
-            raise ValueError("Element out of range: {}".format(value))
+        assert 0 <= value < len(self)
         return self[value]
 
-    def __add__(self, other):
+    def __add__(self, other) -> "Perm":
```

Comment:
I think it's more pythonic to check the type of other and `return NotImplemented` when the class of the object is not a Perm